move/squid/sources/squid/swap_info.move

```diff
@@ -3,102 +3,107 @@ module squid::swap_info;
 use squid::coin_bag::{Self, CoinBag};
 use sui::bcs;
 
+public enum SwapStatus has copy, drop, store {
+    Swapping{index: u64, fallback: bool},
+    Estimating{index: u64, fallback: bool},
+    Done,
+}
+
 public struct SwapInfo {
-    swap_index: u64,
-    estimate_index: u64,
-    status: u8,
+    status: SwapStatus,
     swap_data: vector<vector<u8>>,
     coin_bag: CoinBag,
 }
 
-const ESTIMATING: u8 = 0;
-const SWAPPING: u8 = 1;
-const SKIP_SWAP: u8 = 2;
+#[error]
+const EOutOfEstimates: vector<u8> = b"Trying to get make an estimate but there are none left.";
+#[error]
+const EOutOfSwaps: vector<u8> = b"Trying to get make a swap but there are none left.";
+#[error]
+const ENotEstimating: vector<u8> = b"Trying to get an estimate but estimating is done.";
+#[error]
+const ENotDoneEstimating: vector<u8> = b"Trying to swap while still swapping.";
+#[error]
+const EDoneSwapping: vector<u8> = b"Trying to swap but swapping is finished.";
+#[error]
+const EAlreadySkippingSwaps: vector<u8> = b"Trying to skip swaps while swaps are skipped.";
+#[error]
+const ENotDone: vector<u8> = b"Trying to finalize but SwapInfo is not Done yet.";
+#[error]
+const EDoneEstimating: vector<u8> = b"Trying to estimate but estimating is finished.";
+
 
-const EOutOfEstimates: u64 = 0;
-const EOutOfSwaps: u64 = 1;
-const ENotEstimating: u64 = 3;
-const ENotSwapping: u64 = 4;
-const ENotDoneEstimating: u64 = 5;
-const ENotDoneSwapping: u64 = 6;
 
 public(package) fun new(data: vector<u8>, ctx: &mut TxContext): SwapInfo {
     let swap_data = bcs::new(data).peel_vec_vec_u8();
     SwapInfo {
-        swap_index: 0,
-        estimate_index: 0,
-        status: ESTIMATING,
+        status: SwapStatus::Estimating{index: 0, fallback: false},
         coin_bag: coin_bag::new(ctx),
         swap_data,
     }
 }
 
-public(package) fun get_data_swapping(self: &mut SwapInfo): vector<u8> {
-    let index = self.swap_index;
-    if (index == 0 && self.status == ESTIMATING) {
-        assert!(
-            self.estimate_index == self.swap_data.length(),
-            ENotDoneEstimating,
-        );
-        self.status = SWAPPING;
+public(package) fun get_data_swapping(self: &mut SwapInfo): (vector<u8>, bool) {
```

Comment:
```suggestion
public(package) fun swapping_index(self: &mut SwapInfo): (vector<u8>, bool) {
```
I don't like the use of `data` as a placeholder instead of choosing a more appropriate name. Remove `get_*` from other methods as well

Reply:
swapping index is confusing since it does not return the index, but the bytes.